scipy/special/_basic.py

```diff
@@ -2843,7 +2844,7 @@ def _exact_factorialx_array(n, k=1):
             # e.g. k=3: 26!!! > np.iinfo(np.int32).max
             dt = np.int64
         else:
-            dt = np.int_
+            dt = np_long
```

Comment:
This also looks like "default integer" with the upper branch only being relevant on platforms where the default integer is 32bits.